self-shutdown-route/src/main.rs

```diff
@@ -0,0 +1,56 @@
+use std::{sync::mpsc, thread};
+
+use futures::executor;
+
+use actix_web::{get, middleware, post, web, App, HttpResponse, HttpServer};
+
+#[get("/hello")]
+async fn hello() -> &'static str {
+    "Hello world!"
+}
+
+#[post("/stop")]
+async fn stop(stopper: web::Data<mpsc::Sender<()>>) -> HttpResponse {
+    // make request that sends message through the Sender
+    stopper.send(()).unwrap();
+
+    HttpResponse::NoContent().finish()
+}
+
+#[actix_rt::main]
+async fn main() -> std::io::Result<()> {
+    std::env::set_var("RUST_LOG", "actix_server=debug,actix_web=debug");
+    env_logger::init();
+
+    // create a channel
+    let (tx, rx) = mpsc::channel::<()>();
+
+    let bind = "127.0.0.1:8080";
+
+    // start server as normal but don't .await after .run() yet
+    let server = HttpServer::new(move || {
+        // give the server a Sender in .data
+        let stopper = tx.clone();
+
+        App::new()
+            .data(stopper)
+            .wrap(middleware::Logger::default())
+            .service(hello)
+            .service(stop)
+    })
+    .bind(&bind)?
+    .run();
+
+    // clone the Server handle
+    let srv = server.clone();
```

Comment:
Why do I get : no method named `clone` found for struct `actix_web::dev::Server` in the current scope ?

Reply:
because this example has since been reworked for actix web v4, see the master branch's shutdown-server example